### ISSUE TITLE
[torch.compile][fullgraph] Dynamo rejects comparing `data_ptr()` results (`Unsupported: Builtin operator.* comparison with constant self`)

### ISSUE THREAD

**Issue by tinywisdom** [pytorch/pytorch#165408]
### 🐛 Describe the bug

A tiny model that compares tensor data_ptr()/stride()/shape between an input and its detach() runs fine in eager, but fails under torch.compile(fullgraph=True) in the Dynamo front-end with:
```
torch._dynamo.exc.Unsupported: Builtin `operator.*` comparison with constant `self` failed
Explanation: Failed to compare DataPtrVariable() with DataPtrVariable(), because DataPtrVariable() is not a Python constant or its mutation check fails.
```
This blocks full-graph compilation for otherwise pure-inference code that only inspects metadata and returns a boolean-as-long tensor.

### Minimal Repro
```python
import time, torch
from torch import nn

def env_info():
    print(f"[Env] torch.__version__ = {torch.__version__}")
    print(f"[Env] cuda.is_available = {torch.cuda.is_available()}")
    if torch.cuda.is_available():
        print(f"[Env] torch.version.cuda = {torch.version.cuda}")
        print(f"[Env] torch.backends.cudnn.version() = {torch.backends.cudnn.version()}")

class MyModel(nn.Module):
    def forward(self, x):
        original = x
        detached = x.detach()
        same_data = (
            original.data_ptr() == detached.data_ptr() and
            original.stride() == detached.stride() and
            original.shape == detached.shape
        )
        return torch.tensor([same_data], dtype=torch.long, device=x.device)

def GetInput():
    return torch.rand(1, 1, 1, 3, dtype=torch.float32)

def run_once(model, x, label="run"):
    model.eval()
    if torch.cuda.is_available(): torch.cuda.synchronize()
    t0 = time.perf_counter()
    with torch.no_grad():
        y = model(x)
    if torch.cuda.is_available(): torch.cuda.synchronize()
    print(f"[{label}] output: {y.cpu().tolist()}")
    return y

def main():
    env_info()
    device = "cuda" if torch.cuda.is_available() else "cpu"
    model = MyModel().to(device).eval()
    x = GetInput().to(device)

    # Eager OK
    run_once(model, x, "eager")

    # Compile fullgraph -> fails
    compiled = torch.compile(model, fullgraph=True)
    run_once(compiled, x, "compiled")

if __name__ == "__main__":
    main()
```

### Error logs
```
[eager] output: [1]
...
torch._dynamo.exc.Unsupported: Builtin `operator.*` comparison with constant `self` failed
Explanation: Failed to compare DataPtrVariable() with DataPtrVariable(), because DataPtrVariable() is not a Python constant or its mutation check fails.

from user code:
  original.data_ptr() == detached.data_ptr()
  ...
Set TORCHDYNAMO_VERBOSE=1 / TORCH_LOGS="+dynamo" for full internal trace.
```

### Versions

```
PyTorch version:  2.8.0a0+gitba56102 
Is debug build: True
CUDA used to build PyTorch: 12.6
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04.4 LTS (x86_64)
GCC version: (Ubuntu 12.3.0-1ubuntu1~22.04) 12.3.0
Clang version: Could not collect
CMake version: version 3.22.1
Libc version: glibc-2.35

Python version: 3.10.16 (main, Dec 11 2024, 16:24:50) [GCC 11.2.0] (64-bit runtime)
Python platform: Linux-6.8.0-59-generic-x86_64-with-glibc2.35
Is CUDA available: True
```

cc @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @kadeng @amjames @Lucaskabela @jataylo @chenyang78
